Formula/openjdk.rb

```diff
@@ -1,8 +1,16 @@
 class Openjdk < Formula
   desc "Development kit for the Java programming language"
   homepage "https://openjdk.java.net/"
-  url "https://hg.openjdk.java.net/jdk-updates/jdk15u/archive/jdk-15.0.1-ga.tar.bz2"
-  sha256 "9c5be662f5b166b5c82c27de29b71f867cff3ff4570f4c8fa646490c4529135a"
+  if Hardware::CPU.arm?
```

Comment:
Of note: `zulu` in `cask-versions` does the opposite check, assuming ARM unless Intel. I believe their logic is assume the newest (and eventually to be default) architecture/OS unless proven otherwise.

Is there any consensus on the best way to do these checks for differing Intel/ARM versions?

Reply:
This has no importance

Reply:
@vitorgalvao outlines good reasons for checking in that order for _Casks_ (I can't find the comment anymore), but I don't think these reasons apply to formulae, as the homebrew/core (almost) always uses platform-independent sources.

Reply:
> Of note: `zulu` in `cask-versions` does the [opposite check](https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/zulu8.rb#L2-L14), assuming ARM unless Intel. I believe their logic is assume the newest (and eventually to be default) architecture/OS unless proven otherwise.
> 
> Is there any consensus on the best way to do these checks for differing Intel/ARM versions?

Does `Hardware::CPU.arm?` resolve true for Linux too?  If so,  I'd argue that it should also check OS, since openjdk has had ARM support on Linux for quite some time.  That's moot if [linux doesn't work](https://github.com/Homebrew/homebrew-core/pull/65670/files#diff-df00e346e6e2293ca1363f7b202030ab4e24569f520a49635bc29692e7eb5283R112-R121) but good for formulae authors to know moving forward.

Reply:
Right now, the formula doesn't support linux, and I want to make the smallest change possible.
(And yes, `Hardware::CPU.arm?` would work as expected on linux)

Reply:
> Right now, the formula doesn't support linux, and I want to make the smallest change possible.
> (And yes, `Hardware::CPU.arm?` would work as expected on linux)

I understand this conversation is marked as resolved, but for completeness sake, I tested Linux ARM out and it requires a portable version of Ruby which isn't readily available as precompiled for that platform, so the configuration doesn't have proper support yet and won't be part of the Linux PR either.  I'll try to remember to mention this in the openjdk Linux PR as to explain why this combination isn't accounted for.